pkg/sanity/volume.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2017 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanity
+
+import (
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/api/client"
+	volumeclient "github.com/libopenstorage/openstorage/api/client/volume"
+
+	"github.com/libopenstorage/openstorage/volume"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Volume [Volume Tests]", func() {
+	var (
+		restClient   *client.Client
+		volumedriver volume.VolumeDriver
+	)
+
+	BeforeEach(func() {
+		var err error
+		restClient, err = volumeclient.NewDriverClient(osdAddress, "nfs", volume.APIVersion, "")
+
+		Expect(err).ToNot(HaveOccurred())
+		//volumedriver = clusterclient.ClusterManager(restClient)
+		volumedriver = volumeclient.VolumeDriver(restClient)
+	})
+
+	AfterEach(func() {
+
+	})
+
+	Describe("Volume Create", func() {
+
+		var (
+			volumerequest *api.VolumeCreateRequest
+			volumeID      string
+		)
+
+		BeforeEach(func() {
+
+			var err error
+
+			volumerequest = &api.VolumeCreateRequest{
+				Locator: &api.VolumeLocator{
+					Name: "vol-osd-sanity-cd",
+					VolumeLabels: map[string]string{
+						"class": "sanity-test-class",
+					},
+				},
+				Source: &api.Source{},
+				Spec: &api.VolumeSpec{
+					Size:   1,
+					Format: api.FSType_FS_TYPE_NFS,
+				},
+			}
+
+			volumeID, err = volumedriver.Create(volumerequest.GetLocator(), volumerequest.GetSource(), volumerequest.GetSpec())
```

Comment:
Normally the action to be tested is to be included in the It() test. For example, sometimes you may want to pass in correct arguments, and sometimes you may want to pass in incorrect arguments expecting failure.

But it is ok to do this _if_ you there is no negative tests.